internal/kafka/internal/api/dbapi/kafka_request_types.go

```diff
@@ -141,3 +143,67 @@ func (k *KafkaRequest) GetExpirationTime(lifespanSeconds int) *time.Time {
 	expireTime := k.CreatedAt.Add(time.Duration(lifespanSeconds) * time.Second)
 	return &expireTime
 }
+
+// The RemainingLifeSpan is the remaining life a kafka instance have before it expires and gets deleted
+type RemainingLifeSpan interface {
+	// Infinite returns whether the Kafka instance has infinite (no expiration) remaining life span
+	Infinite() bool
+
+	// Days returns the number of remaining life span days a Kafka instance has. If infinite, returns -1
+	Days() float64
+}
+
+var _ RemainingLifeSpan = &remainingLifeSpan{}
+
+// remainingLifeSpan is an implementation of the RemainingLifeSpan interface
+type remainingLifeSpan struct {
+	// remainingLifeSpanDays is the number of days before the Kafka instance will be deleted. It is a `float64` to be able to manage fraction of days.
+	remainingLifeSpanDays float64
+	// infinite represent whether the Kafka instance has an expiration
+	infinite bool
+}
+
+func (l *remainingLifeSpan) Infinite() bool {
+	return l.infinite
+}
+
+func (l *remainingLifeSpan) Days() float64 {
+	return l.remainingLifeSpanDays
+}
+
+// IsExpired returns whether a kafka is expired and how many days the instance will live before expiring (-1 never expires)
+func (k *KafkaRequest) IsExpired() (bool, RemainingLifeSpan) {
+	if !k.ExpiresAt.Valid {
+		return false, &remainingLifeSpan{
+			remainingLifeSpanDays: -1,
+			infinite:              true,
+		}
+	}
+
+	now := time.Now()
+	if now.After(k.ExpiresAt.Time) {
+		return true, &remainingLifeSpan{
+			remainingLifeSpanDays: 0,
+			infinite:              false,
+		}
+	}
+
+	remainingLife := k.ExpiresAt.Time.Sub(now)
+	return false, &remainingLifeSpan{
+		remainingLifeSpanDays: remainingLife.Hours() / 24,
+		infinite:              false,
+	}
+}
+
+// CanBeSuspended returns whether the kafka instance can be suspended or not
+func (k *KafkaRequest) CanBeSuspended() bool {
```

Comment:
Maybe we should rename this to `CanBeAutomaticallySuspended` to differentiate it with a manual suspension? The set of states would be different